src/main/java/eu/tsystems/mms/tic/testerra/plugins/xray/synchronize/AbstractXrayResultsSynchronizer.java

```diff
@@ -221,22 +221,29 @@ private synchronized void flushSyncQueue() {
         boolean areNewTestsToImport = testSetSyncQueue.stream().anyMatch(xrayTestSetIssue -> {
             return xrayTestSetIssue.getTestKeys().stream().anyMatch(key -> key.contains(XrayUtils.PREFIX_NEW_ISSUE));
         });
-        if (areNewTestsToImport) {
-            xrayTestExecutionImport.getResultTestIssueImport().getSuccess().forEach(jiraIssueReference -> {
-                try {
-                    // Replace the temporary key with real Jira key from the result 'xrayTestExecutionImport'
-                    JiraIssue issue = xrayUtils.getIssue(jiraIssueReference.getKey());
-                    testSetSyncQueue.forEach(xrayTestSetIssue -> {
-                        Optional<String> findKey = xrayTestSetIssue.getTestKeys().stream().filter(key -> key.contains(issue.getSummary())).findFirst();
-                        if (findKey.isPresent()) {
-                            xrayTestSetIssue.getTestKeys().removeIf(key -> key.contains(issue.getSummary()));
-                            xrayTestSetIssue.getTestKeys().add(issue.getKey());
-                        }
-                    });
-                } catch (IOException e) {
-                    log().error(String.format("Unable to read %s", IssueType.TestExecution), e);
-                }
-            });
+        if (areNewTestsToImport && xrayTestExecutionImport.getResultTestIssueImport() != null) {
```

Comment:
This seems to be an unfortunate weakening of the (internal?) API: previously `getResultTestIssueImport` was not allowed to return `null` but now it is. ... Question is: Can that not be forbidden?

Reply:
It cannot forbidden. The content based on an API call, see https://github.com/telekom/testerra-xray-connector/blob/919197b0e7db170eb87ddae91af0dea8ca656ad1/src/main/java/eu/tsystems/mms/tic/testerra/plugins/xray/util/XrayUtils.java#L55

In case of an exception at the POST request there is no chance to fill `xrayTestExecutionImport`. An exception could occure if I try to send invalid requests, e.g. with invalid values.